data/crac/crac-result-extensions/src/test/java/com.farao_community.farao.data.crac_result_extensions/RangeActionResultTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.data.crac_result_extensions;
+
+import com.farao_community.farao.data.crac_api.Instant;
+import com.farao_community.farao.data.crac_api.State;
+import com.farao_community.farao.data.crac_impl.ComplexContingency;
+import com.farao_community.farao.data.crac_impl.SimpleState;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Joris Mancini {@literal <joris.mancini at rte-france.com>}
+ */
+public class RangeActionResultTest {
+    private static final double EPSILON = 0.01;
+
+    private RangeActionResult rangeActionResult;
+    private Set<State> states;
+    private State initialState;
+    private State outage1;
+    private State curative1;
+    private State outage2;
+    private State curative2;
+
+    @Before
+    public void setUp() {
+        states = new HashSet<>();
+        initialState = new SimpleState(Optional.empty(), new Instant("initial", 0));
+        outage1 = new SimpleState(Optional.of(new ComplexContingency("co1")), new Instant("after-co1", 10));
+        curative1 = new SimpleState(Optional.of(new ComplexContingency("co1")), new Instant("curative-co1", 50));
+        outage2 = new SimpleState(Optional.of(new ComplexContingency("co2")), new Instant("after-co2", 10));
+        curative2 = new SimpleState(Optional.of(new ComplexContingency("co2")), new Instant("curative-co2", 50));
+        states.add(initialState);
+        states.add(outage1);
+        states.add(curative1);
+        states.add(outage2);
+        states.add(curative2);
+        rangeActionResult = new RangeActionResult(states);
+    }
+
+    @Test
+    public void constructor() {
+        assertTrue(rangeActionResult.activationMap.containsKey(initialState));
+        assertTrue(rangeActionResult.activationMap.containsKey(outage1));
+        assertTrue(rangeActionResult.activationMap.containsKey(curative1));
+        assertTrue(rangeActionResult.activationMap.containsKey(outage2));
+        assertTrue(rangeActionResult.activationMap.containsKey(curative2));
+        assertEquals(5, rangeActionResult.activationMap.size());
+    }
+
+    @Test
+    public void getSetPoint() {
+        rangeActionResult.activate(initialState);
+        assertTrue(rangeActionResult.isActivated(initialState));
+        assertFalse(rangeActionResult.isActivated(outage1));
+        rangeActionResult.setSetPoint(outage1, 15.);
+        assertEquals(Double.NaN, rangeActionResult.getSetPoint(initialState), EPSILON);
```

Comment:
I don't think we want such a behaviour. It makes no sense to activate a RangeAction with a setpoint equals to Double.NaN does it?
I suggest the following refactoring: 
1) remove the activate method from the abstract class and implement it into the two concrete class. 
public void activate(State state) in NetworkActionResult
public void activate(State state, double setpoint) in RangeActionResult
2) override the deactivate method in RangeActionResult to reset the value of the setpointMap to Double.NaN